ludwig/models/llm.py

```diff
@@ -407,6 +420,8 @@ def load(self, save_path):
             weights_save_path = os.path.join(save_path, MODEL_WEIGHTS_FILE_NAME)
             config = PeftConfig.from_pretrained(weights_save_path)
             config.inference_mode = False
+            ###SXK I noticed we are tightly coupling to hugging face in places, I wonder
+            ###if we want to make this part configurable as well
```

Comment:
It is true that we are tighly coupling, but I'm not sure it's worth doing that today since. We could make this a registry pattern where we load LLMs different based on their "types", just like Ludwig does today for a lot of its tokenizers. Additionally, most open source large language models are either on huggingface or on github (and then eventually make their way to huggingface anyway), so this being tightly coupled seems fine for now.

In the future, I do think your suggestion of having a base LLM class and this being a subclass of that called HFLLM or something could make sense though!